File: src/main/java/bank/Deposit.java
package bank;

import java.time.LocalDate;
import java.util.Optional;

public class Deposit extends Transaction {

	public Deposit(float amount, LocalDate dateTime) {
		super(amount, dateTime);
	}

	@Override
	public Optional<Float> getDebit() {
		return Optional.empty();
	}

	@Override
	public Optional<Float> getCredit() {
		return Optional.of(amount);
	}

}

File: scala/src/main/scala/domain/Deposit.scala
package bank.domain

import java.util.Date

class Deposit(val amount: Double, operation: Date) extends Transaction(operation){
  if(amount <= 0) throw new IllegalArgumentException("the number must be strictly positive.")

  override def deposit: Double = amount

  override def withdraw: Double = 0

  override def equals(that: Any): Boolean =
    that match {
      case that: Deposit => this.amount.equals(that.amount) && this.operation.equals(that.operation)
      case _ => false
   }

  override def hashCode:Int = {
    val prime = 31
    var result = 1
    result = prime * result + amount.hashCode;
    result = prime * result + operation.hashCode;
    return result
  }

  override def toString = s"Deposit of $amount"

}
